server/src/main/java/io/crate/expression/scalar/regex/RegexpMatchScalar.java

```diff
@@ -0,0 +1,129 @@
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public class RegexpMatchScalar extends Scalar<List<String>, String> {
+    private Pattern pattern;
+    public static final String NAME = "regexp_match";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING_ARRAY.getTypeSignature()
+            ),
+            RegexpMatchScalar::new
+        );
+
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING_ARRAY.getTypeSignature()
+            ),
+            RegexpMatchScalar::new
+        );
+    }
+
+    public RegexpMatchScalar(Signature signature, BoundSignature boundSignature) {
+        super(signature, boundSignature);
+    }
+
+    public RegexpMatchScalar(Signature signature, BoundSignature boundSignature, String patternString, int flags) throws PatternSyntaxException {
+        super(signature, boundSignature);
+        compilePattern(patternString, flags);
+    }
+
+    private void compilePattern(String patternString, int flags) throws PatternSyntaxException {
+        this.pattern = Pattern.compile(patternString, flags);
+    }
+
+
+    @Override
+    public List<String> evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<String>[] args) throws IllegalArgumentException{
+        assert args.length == 2 || args.length == 3 : "number of args must be 2 or 3";
+        String source = args[0].value();
+        if (source == null) {
+            return null;
+        }
+
+        String pattern = args[1].value();
+        if (pattern == null) {
+            return null;
+        }
+
+        String flags = null;
+
+        if (args.length == 3) {
+            flags = args[2].value();
+            if (!isGlobal(flags)) {
+                pattern = Pattern.compile(pattern, parseFlags(flags)).pattern();
+            } else{
+                throw new IllegalArgumentException("Unsupported flag: 'g'. Global flag is not supported.");
+            }
+        }
+
+        List<String> matches = new ArrayList<>();
+        java.util.regex.Matcher matcher = java.util.regex.Pattern.compile(pattern).matcher(source);
+        while (matcher.find()) {
+            matches.add(matcher.group());
+        }
+        return matches;
+    }
+
+    @Override
+    public Query toQuery(Reference ref, Literal<?> literal) throws IllegalArgumentException {
+        String pattern = (String) literal.value();
+        int flags = 0;
+
+        // Check if the pattern contains flags
+        String flagsStr = null;
+        int flagSeparatorIndex = pattern.lastIndexOf('/');
```

Comment:
I wonder where the flags are added to the pattern by the `/` separator? I think the existing `toQuery` implementations won't work as they only support 2 arguments, a reference and a literal, but not 3.